langchain4j-core/src/main/java/dev/langchain4j/web/search/WebSearchRequest.java

```diff
@@ -0,0 +1,312 @@
+ * <p>
+ * {@link WebSearchRequest} follow opensearch foundation standard implemented by most web search engine libs like Google, Bing, Yahoo, etc.
+ * <a href="https://github.com/dewitt/opensearch/blob/master/opensearch-1-1-draft-6.md#opensearch-11-parameters">OpenSearch#parameters</a>
+ * </p>
+ * <p>
+ * The {@link #searchTerms} are the keywords that the search client desires to search for. This param is mandatory to perform a search.
+ * <p>
+ * <br>
+ * Configurable parameters (optional):
+ * <ul>
+ *    <li>{@link #maxResults} - The expected number of results to be found if the search request were made. Each search engine may have a different limit for the maximum number of results that can be returned.</li>
+ *    <li>{@link #language} - The desired language for search results is a string that indicates that the search client desires search results in the specified language. Each search engine may have a different set of supported languages.</li>
+ *    <li>{@link #geoLocation} - The desired geolocation for search results is a string that indicates that the search client desires search results in the specified geolocation. Each search engine may have a different set of supported geolocations.</li>
+ *    <li>{@link #startPage} - The start page number for search results is the page number of the set of search results desired by the search user.</li>
+ *    <li>{@link #startIndex} - The start index for search results is the index of the first search result desired by the search user. Each search engine may have a different set of supported start indexes in combination with the start page number.</li>
+ *    <li>{@link #safeSearch} - The safe search flag is a boolean that indicates that the search client desires search results with safe search enabled or disabled.</li>
+ *    <li>{@link #additionalParams} - The additional parameters for the search request are a map of key-value pairs that represent additional parameters for the search request. It's a way to be flex and add custom param for each search engine.</li>
+*/
+public class WebSearchRequest {
+
+    private final String searchTerms;
+    private final Integer maxResults;
+    private final String language;
+    private final String geoLocation;
+    private final Integer startPage;
+    private final Integer startIndex;
+    private final Boolean safeSearch;
+    private final Map<String, Object> additionalParams;
+
+    private WebSearchRequest(Builder builder){
+        this.searchTerms = ensureNotBlank(builder.searchTerms,"searchTerms");
+        this.maxResults = builder.maxResults;
+        this.language = builder.language;
+        this.geoLocation = builder.geoLocation;
+        this.startPage = getOrDefault(builder.startPage,1);
+        this.startIndex = builder.startIndex;
+        this.safeSearch = getOrDefault(builder.safeSearch,true);
+        this.additionalParams = getOrDefault(builder.additionalParams, new HashMap<>());
```

Comment:
```suggestion
        this.additionalParams = getOrDefault(builder.additionalParams, () -> new HashMap<>());
```
minor: to avoid creating a map each time a search is done